File: checkIns/src/main/java/com/popcorn/checkIns/event/standard/EventLineItem.java
package com.popcorn.checkIns.event.standard;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * 표준 이벤트의 주문 라인 아이템
 * 주문 내 개별 항목들을 표현
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class EventLineItem {

    /**
     * 아이템 타입 (RESERVATION, GOODS)
     */
    private String itemType;

    /**
     * 스케줄 ID (예약 아이템인 경우)
     */
    private UUID scheduleId;

    /**
     * 굿즈 변형 ID (굿즈 아이템인 경우)
     */
    private UUID goodsId;

    /**
     * 수량
     */
    private Integer qty;

    /**
     * 단가
     */
    private BigDecimal unitPrice;

    /**
     * 라인 총액 (단가 * 수량)
     */
    private BigDecimal linePrice;

    /**
     * 아이템명 (optional)
     */
    private String itemName;

    /**
     * 아이템 설명 (optional)
     */
    private String itemDescription;
}

File: payment/src/main/kotlin/com/popcorn/payment/event/standard/EventLineItem.kt
package com.popcorn.payment.event.standard

import java.util.UUID

/**
 * 표준 이벤트 라인 아이템 구조 (Kotlin 버전)
 * 주문 내 개별 항목 정보 (예약/굿즈)
 */
data class EventLineItem(
    /**
     * 아이템 타입: SCHEDULE(예약) 또는 GOODS(굿즈)
     */
    val itemType: String? = null,

    // === 예약 관련 필드 (itemType = "SCHEDULE") ===
    /**
     * 스케줄 ID (예약용)
     */
    val scheduleId: UUID? = null,

    /**
     * 스케줄 시작 시간
     */
    val scheduleStartAt: String? = null,

    /**
     * 스케줄 종료 시간
     */
    val scheduleEndAt: String? = null,

    // === 굿즈 관련 필드 (itemType = "GOODS") ===
    /**
     * 굿즈 변형 ID (굿즈용)
     */
    val goodsVariantId: UUID? = null,

    /**
     * 굿즈 이름
     */
    val goodsName: String? = null,

    /**
     * 재고 단위 (예: "BLACK-M")
     */
    val stockUnit: String? = null,

    // === 공통 필드 ===
    /**
     * 주문 굿즈 ID
     */
    val orderGoodsId: UUID? = null,

    /**
     * 수량
     */
    val qty: Int? = null,

    /**
     * 단가
     */
    val unitPrice: Int? = null,

    /**
     * 라인 총 금액 (unitPrice * qty)
     */
    val linePrice: Int? = null
)